File: SEGUNDO/PDOO/PRACTICAS/PRACTICA1/WeaponType.rb
#encoding:utf-8

module Deepspace
	module WeaponType
		class Type
			def initialize (p) #"constructor"
				@power=p
			end
			
			public
			def power
				@power
			end
		end
		
		LASER = Type.new(2.0)
		MISSILE = Type.new(3.0)
		PLASMA = Type.new(4.0)
	end
end

File: SEGUNDO/PDOO/PRACTICAS/PRACTICA2/java/deepSpace/src/deepspace/WeaponType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author lauralazaro
 */
public enum WeaponType {
    LASER (2.0f),
    MISSILE (3.0f), 
    PLASMA (4.0f);
    
    private final float power;
    
    WeaponType (float p) {
        this.power = p;
    }
    
    float getPower() {
        return power;
    }
}
